### ISSUE TITLE
[Enhancement]: Implement Consistent Template Handling for Order Confirmation in WooCommerce

### ISSUE THREAD

**Issue by chvillanuevap** [woothemes/woocommerce#43601]
### Describe the solution you'd like

We currently have the ability to override templates for the Cart and Checkout pages using `page-cart.html` and `page-checkout.html` (or `cart.html` and `checkout.html` for legacy support). However, this flexibility does not extend to the Order Confirmation page. While there is an `OrderConfirmationTemplate` class available, it lacks a corresponding template or page in the Site Editor. This omission makes it confusing as to what is the best practice for customizing this template.

Furthermore, it appears that this template can be overridden using a `order-confirmation.html` file, but this method seems inconsistent with the WordPress template hierarchy (similar to the old `cart.html` and `checkout.html` templates). I would like to suggest having a standardized way for handling the Order Confirmation template that aligns with WordPress' best practices. This could involve creating a new template file (like `page-order-confirmation.html`) or integrating more closely with the WordPress template hierarchy. 

### Describe alternatives you've considered

As a temporary solution, I have created a `order-confirmation.html` file in my templates directory. This file redirects to the Checkout page, which continues to use the old `woocommerce_checkout` shortcode. This shortcode is necessary due to its hooks and filters, which are utilized by older plugins. Here's a snippet of the current implementation:
```
<!-- wp:template-part {"slug":"header","theme":"thepostmansknock","tagName":"header","className":"site-header"} /-->
<!-- wp:group {"tagName":"main","className":"site-content","style":{"spacing":{"margin":{"top":"0","bottom":"0"},"padding":{"top":"2.5rem","bottom":"3rem"}}}} -->
<main class="wp-block-group site-content" style="margin-top:0;margin-bottom:0;padding-top:2.5rem;padding-bottom:3rem">
	<!-- wp:woocommerce/page-content-wrapper {"page":"checkout"} -->
		<!-- wp:group {"tagName":"article","className":"entry"} -->
		<article class="wp-block-group entry">
			<!-- wp:group {"tagName":"header","layout":{"type":"constrained"},"className":"entry-header"} -->
			<header class="wp-block-group entry-header">
				<!-- wp:post-title {"textAlign":"center","level":1} /-->
			</header>
			<!-- /wp:group -->
			<!-- wp:post-content {"layout":{"type":"constrained"},"className":"entry-content"} /-->
		</article>
		<!-- /wp:group -->
	<!-- /wp:woocommerce/page-content-wrapper -->
</main>
<!-- /wp:group -->
<!-- wp:template-part {"slug":"footer","theme":"thepostmansknock","tagName":"footer","className":"site-footer"} /-->
```

**Comment by Aljullu** [comment]
Thanks for opening this issue, @chvillanuevap! From my understanding, you would like themes to be able to override the _Order Confirmation_ page using a `page-order-confirmation.html` file instead of `order-confirmation.html`, is that correct? Prefixing the template file with `page-` would make it consistent with `page-cart.html` and `page-checkout.html`,  but besides that, would there any other benefit of it?

**Comment by chvillanuevap** [comment]
Yes, you're correct. The main benefits of adopting a `page-order-confirmation.html` naming convention is the consistency, as it aligns with the naming convention of other page templates like `page-cart.html` and `page-checkout.html`, making the theme structure easier to understand. It would also help developers, like myself, quickly identify and customize the Order Confirmation page, improving workflow and reducing confusion. It would have definitely saved me some work time!

**Comment by kkoteen** [comment]
Hi @chvillanuevap - thanks for such a detailed report, it's really helpful.

I know this has been open for a long time, and I appreciate your patience. We're currently going through some older WooCommerce issues, identifying what still needs fixing, and tidying up what's already fixed or out of scope.

I'd like to get an engineer's eyes on this before we make a call, so I've flagged it for developer feedback. 

@Aljullu can you take a closer look and advise? Thanks!

**Comment by Aljullu** [comment]
While I recognize it's a bit inconsistent that Cart and Checkout use the `page-` prefix while Order Confirmation doesn't, there are other templates that don't have the `page-` prefix either, like `archive-product.html` and `coming-soon.html`. So I would probably lean more towards removing the `page-` prefix than adding it to other templates.

Having said that, there are currently some ongoing investigations towards improving and clarifying the relation between pages and templates (https://github.com/woocommerce/woocommerce/issues/42152), so I think we should leave this as-is for now.

**Comment by kkoteen** [comment]
Thanks for your input, @Aljullu! I’m going to close this issue, as this work isn’t currently planned.

@chvillanuevap, thank you for taking the time to submit this request. We’ll keep your suggestion in mind if we revisit templates in the future.
